File: common/src/main/java/org/bread_experts_group/breadlib/extensions/item/IKeyboardItem.java
package org.bread_experts_group.breadlib.extensions.item;

import net.minecraft.client.multiplayer.ClientLevel;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;

public interface IKeyboardItem {
	void onKeyPress(
			int button,
			int scanCode,
			int action,
			int modifiers,
			ItemStack heldStack,
			ClientLevel level,
			Player player
	);
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/extensions/item/IKeyboardItem.kt
package org.bread_experts_group.breadlib.extensions.item

import net.minecraft.client.multiplayer.ClientLevel
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack

interface IKeyboardItem {
	fun onKeyPress(
		button: Int,
		scanCode: Int,
		action: Int,
		modifiers: Int,
		heldStack: ItemStack,
		level: ClientLevel,
		player: Player
	)
}